backend/handlers/bets/betutils/feeutils_test.go

```diff
@@ -0,0 +1,198 @@
+					SellSharesFee int64 `yaml:"sellSharesFee"`
+				} `yaml:"betFees"`
+			} `yaml:"betting"`
+		}{
+			MarketCreation: struct {
+				InitialMarketProbability   float64 `yaml:"initialMarketProbability"`
+				InitialMarketSubsidization int64   `yaml:"initialMarketSubsidization"`
+				InitialMarketYes           int64   `yaml:"initialMarketYes"`
+				InitialMarketNo            int64   `yaml:"initialMarketNo"`
+			}{
+				InitialMarketProbability:   0.5,
+				InitialMarketSubsidization: 10,
+				InitialMarketYes:           0,
+				InitialMarketNo:            0,
+			},
+			MarketIncentives: struct {
+				CreateMarketCost int64 `yaml:"createMarketCost"`
+				TraderBonus      int64 `yaml:"traderBonus"`
+			}{
+				CreateMarketCost: 10,
+				TraderBonus:      1,
+			},
+			User: struct {
+				InitialAccountBalance int64 `yaml:"initialAccountBalance"`
+				MaximumDebtAllowed    int64 `yaml:"maximumDebtAllowed"`
+			}{
+				InitialAccountBalance: 1000,
+				MaximumDebtAllowed:    500,
+			},
+			Betting: struct {
+				MinimumBet int64 `yaml:"minimumBet"`
+				BetFees    struct {
+					InitialBetFee int64 `yaml:"initialBetFee"`
+					BuySharesFee  int64 `yaml:"buySharesFee"`
+					SellSharesFee int64 `yaml:"sellSharesFee"`
+				} `yaml:"betFees"`
+			}{
+				MinimumBet: 1,
+				BetFees: struct {
+					InitialBetFee int64 `yaml:"initialBetFee"`
+					BuySharesFee  int64 `yaml:"buySharesFee"`
+					SellSharesFee int64 `yaml:"sellSharesFee"`
+				}{
+					InitialBetFee: 1,
+					BuySharesFee:  0,
+					SellSharesFee: 0,
+				},
+			},
+		},
+	}
+}
+
+func TestGetUserInitialBetFee(t *testing.T) {
+	// Set up in-memory SQLite database
+	db, err := gorm.Open(sqlite.Open(":memory:"), &gorm.Config{})
+	if err != nil {
+		t.Fatalf("Failed to connect to the database: %v", err)
+	}
+
+	// Migrate the Bet and User models
+	db.AutoMigrate(&models.Bet{}, &models.User{})
+
+	// Mock the appConfig with test data
+	appConfig = mockEconomicConfig()
+
+	// Create a test user with a unique api_key
+	user := &models.User{
+		Username:       "testuser",
+		AccountBalance: 1000,
+		ApiKey:         "unique_api_key_1", // Ensure this is unique
+	}
+	db.Create(&user) // Save the user to the database
+
+	// Scenario 1: User places a bet on Market 1 where they have no prior bets
+	initialBetFee := getUserInitialBetFee(db, 1, user)
+	if initialBetFee != appConfig.Economics.Betting.BetFees.InitialBetFee {
+		t.Errorf("Expected initial bet fee to be %d, got %d", appConfig.Economics.Betting.BetFees.InitialBetFee, initialBetFee)
+	}
+
+	// Place a bet for the user on Market 1
+	bets := []models.Bet{
+		{Username: "testuser", MarketID: 1, Amount: 100, PlacedAt: time.Now()},
+	}
+	db.Create(&bets) // Save the bet to the database
+
+	// Scenario 2: User places another bet on Market 1 where they already have a bet
+	initialBetFee = getUserInitialBetFee(db, 1, user)
+	if initialBetFee != 0 {
+		t.Errorf("Expected initial bet fee to be 0, got %d", initialBetFee)
```

Comment:
https://go.dev/wiki/TestComments#got-before-want recommends switching order of test output to got before want.